### ISSUE TITLE
GPUPersistentKernel fails to validate sdfg with local variables

### ISSUE THREAD

**Issue by kylosus** [spcl/dace#1304]
**Describe the bug**
`GPUPersistentKernel` transformation incorrectly considers symbols defined within the SDFG as free symbols and fails to validate.

**To Reproduce**
Consider the following toy example:
```python
import dace as dc
import numpy as np

from dace.sdfg.graph import SubgraphView
from dace.transformation.subgraph import GPUPersistentKernel

import dace.transformation.transformation

N = dc.symbol('N', dtype=dc.int64)


@dc.program(auto_optimize=False, device=dace.DeviceType.GPU)
def func(A: dc.float64[N], B: dc.float64[N]):
    a = 10.2

    for t in range(1, 10000):
        if t < N:
            A[:] = (A + B + a) / 2
            a += 1


if __name__ == "__main__":
    # Initialization
    N = 10000
    A = np.random.rand(N)
    B = np.random.rand(N)

    sdfg = func.to_sdfg()
    sdfg.apply_gpu_transformations()
    content_nodes = set(sdfg.nodes()) - {sdfg.start_state, sdfg.sink_nodes()[0]}
    subgraph = SubgraphView(sdfg, content_nodes)

    transform = GPUPersistentKernel()
    transform.setup_match(subgraph)
    transform.kernel_prefix = 'stuff'
    transform.apply(sdfg)

    func = sdfg.compile()
    func(A=A, B=B, N=N)
```
Attempting to call this dace program fails with the following error:
```
Traceback (most recent call last):
  File ".../jacobi_test_2.py", line 49, in <module>
    func(A=A, B=B, N=N)
  File ".../dace/codegen/compiled_sdfg.py", line 334, in __call__
    argtuple, initargtuple = self._construct_args(kwargs)
  File ".../dace/dace/codegen/compiled_sdfg.py", line 395, in _construct_args
    raise KeyError("Missing program argument \"{}\"".format(a))
KeyError: 'Missing program argument "a"'
```
Adding extra `a=0` and `t=0` args allows the sdfg to pass the validation and runs as expected.

**Desktop (please complete the following information):**
 - Python 3.10.12
 - Dace latest master branch

**Additional context**
Changing [this](https://github.com/spcl/dace/blob/5d66e936bd2fe198dd1f23bdfcf090842dfc3619/dace/transformation/subgraph/gpu_persistent_fusion.py#L197) line to the following fixes it but I'm not sure if it's a full solution.
```py
sdfg.add_edge(launch_state, exit_state_out, InterstateEdge(None, entry_edge.data.assignments))
```

**Comment by tbennun** [comment]
Thank you for reporting! After looking into the issue, it looks like the transformation did not remove the symbols that are now unused from the outer SDFG. This is now fixed in #1322.

**Comment by tbennun** [comment]
@kylosus The PR was merged, please let us know if this fixes your issue.
